homeassistant/components/reolink/update.py

```diff
@@ -68,6 +68,25 @@ def latest_version(self) -> str | None:
 
         return self.coordinator.data.version_string
 
+    @property
+    def supported_features(self) -> UpdateEntityFeature:
+        """Flag supported features."""
+        supported_features = UpdateEntityFeature.INSTALL
+        if isinstance(self.coordinator.data, NewSoftwareVersion):
+            supported_features |= UpdateEntityFeature.RELEASE_NOTES
+        return supported_features
+
+    async def async_release_notes(self) -> str | None:
+        """Return the release notes."""
+        if not isinstance(self.coordinator.data, NewSoftwareVersion):
+            return None
+
+        return (
+            f"Directly download the [firmware zip file here]({self.coordinator.data.download_url})"
+            ", including installation guide.\n\n"
```

Comment:
@frenck the build in firmware update utility of Reolink is notoriously bad (and therefore also the API commands).
Often there is a firmware update available from the reolink servers, but when checking on the camera itself it will display that it is already running the latest firmware, same thing can happen when pressing the install new firmware.

However manually uploading the new firmware file to the camera always works (Reolink support also tells people to just check the reolink.com server and update manually instead of them fixing the build in auto firmware update of the camera).

So in case the install button fails (unfortunatlly happens on quite a lot of Reolink models, they should really fix that), the link will directly download a .zip file from the reolink.com server which includes 2 files: the .pak firmware file with the new firmware version and a .pdf file with the instructions on how to install it (just go to the reolink client->settings->system->maintenance->Firmware update, use "browse" button to select the .pak file and hit "update".
The .pdf includes screenshots of where to find these buttons/tabs.

This .zip file is exactly the same file as would be downloaded when using the Reolink Download Center https://reolink.com/download-center/ (the underlying donwload_url is obtained from the same json servers also used by the Reolink Download Center)

So in principle the install button schould work, but often fails, the link provides a alternative.
I will change the sentence to reflect this.